OpenRA.Mods.Common/Traits/Cargo.cs

```diff
@@ -402,29 +402,32 @@ public void Load(Actor self, Actor a)
 		}
 
 		void INotifyKilled.Killed(Actor self, AttackInfo e)
-		{
-			if (Info.EjectOnDeath)
-				while (!IsEmpty() && CanUnload(BlockedByActor.All))
+ 		{
+			if (Info.EjectOnDeath && self.IsAtGroundLevel() && (!checkTerrainType || Info.UnloadTerrainTypes.Contains(self.World.Map.GetTerrainInfo(self.Location).Type)))
```

Comment:
Does this ground level check prevent pilots/paratroopers/etc from being ejected from aircraft?

Reply:
Yes, I'm just keeping the check

As for pilots, `EjectOnDeath` trait exists for them. Though that trait doesn't check for cell occupance